src/org/deepsymmetry/bytefield/core.cljs

```diff
@@ -531,57 +531,85 @@
   the standard `:box-above` style. You can change that by passing
   different attributes under the `:box-above-style` key (for example,
   use `{:box-above-style :box-above-related}` if the gap relates to
-  the preceding box)."
+  the preceding box).
+  
+  The background can be filled with a color passed with `:fill`."
   ([]
    (draw-gap nil nil))
   ([label]
    (draw-gap label nil))
   ([label attr-spec]
-   (let [{:keys [height gap edge gap-style box-above-style min-label-columns]
+   (let [attrs (eval-attribute-spec attr-spec)]
+   (let [{:keys [height gap edge gap-style box-above-style min-label-columns fill]
           :or   {height            70
                  gap               10
                  edge              15
                  gap-style         (eval-attribute-spec :dotted)
                  box-above-style   (eval-attribute-spec :box-above)
-                 min-label-columns 8}} (eval-attribute-spec attr-spec)
+                 min-label-columns 8}} attrs
 
          column (:column @@('diagram-state @*globals*))
          boxes  @('boxes-per-row @*globals*)]
      (if label
        ;; We are supposed to draw a label.
        (if (<= min-label-columns (- boxes column))
-         (draw-box label [{:span (- boxes column)} box-above-style]) ; And there is room for it on the current line.
+         (draw-box label [{:span (- boxes column)} box-above-style (if (nil? fill) {} {:fill fill})]) ; And there is room for it on the current line.
```

Comment:
@brunchboy I'm not a Clojure programmer, so if theres a cleaner way to apply these conditionals please suggest :)

Reply:
I’ll have to re-learn how my weird little styling DSL works here, because I think it’s not just ordinary Clojure, and that could  affect what the best answer is. Typically Clojure does something called nil-punning, though, so `nil` is treated as an empty map in contexts that expect a map, so your conditional might simply be able to be:
```clojure
(when fill {:fill fill})
```
Which will either become that map, or `nil`, which would be treated the same as the `{}` in your conditional.

My afternoon and evening devolved into chaos, though, so I won’t have time to study this until later this weekend. I will pull down your branch and play some then though!

Reply:
OK, I have verified that this approach, taking advantage of nil-punning, works, and simplifies the conditional sum. Since it was used in multiple places, I also added a `let` binding for the result in the consolidated `let` statement above:
```clojure
    fill-style (when fill {:fill fill})
```
Then the individual `draw-box` invocations can just use `fill-style` in place of your `if` expressions.

Reply:
Thanks, I've addressed that in 60253e3